packages/braid-design-system/src/lib/components/private/Modal/ModalContent.tsx

```diff
@@ -99,12 +100,15 @@ export const ModalContent = ({
   data,
   ...restProps
 }: ModalContentProps) => {
+  const fallbackId = useId();
+  const resolvedId = id || fallbackId;
+
   const defaultModalRef = useRef<HTMLElement>(null);
   const modalRef = modalRefProp || defaultModalRef;
   const defaultHeadingRef = useRef<HTMLElement>(null);
   const headingRef = headingRefProp || defaultHeadingRef;
 
-  const descriptionId = `${id}_desc`;
+  const descriptionId = `${resolvedId}_desc`;
```

Comment:
As we did on the other PR, i'd suggest removing the custom suffixes for `descriptionId` and the close button, in favour of their own generated ids